File: .taira-source-c80bf4d85a97/java/norito_java/src/main/java/org/hyperledger/iroha/norito/TypeAdapter.java
// Copyright 2024 Hyperledger Iroha Contributors
// SPDX-License-Identifier: Apache-2.0

package org.hyperledger.iroha.norito;

/** Interface implemented by codec adapters. */
public interface TypeAdapter<T> {
  void encode(NoritoEncoder encoder, T value);

  T decode(NoritoDecoder decoder);

  /** Returns the fixed size in bytes, or -1 if variable. */
  default int fixedSize() {
    return -1;
  }

  /** Returns true if the adapter carries its own length delimiters. */
  default boolean isSelfDelimiting() {
    return false;
  }
}

File: .taira-source-c80bf4d85a97/kotlin/core-jvm/src/main/java/org/hyperledger/iroha/sdk/norito/TypeAdapter.kt
// Copyright 2024 Hyperledger Iroha Contributors
// SPDX-License-Identifier: Apache-2.0

package org.hyperledger.iroha.sdk.norito

/** Interface implemented by codec adapters. */
interface TypeAdapter<T> {
    fun encode(encoder: NoritoEncoder, value: T)

    fun decode(decoder: NoritoDecoder): T

    /** Returns the fixed size in bytes, or -1 if variable. */
    fun fixedSize(): Int = -1

    /** Returns true if the adapter carries its own length delimiters. */
    fun isSelfDelimiting(): Boolean = false
}
